containers/ddev-dbserver/files/create_base_db.sh

```diff
@@ -71,24 +73,24 @@ mysql -uroot <<EOF
 	FLUSH TABLES;
 EOF
 
-mysqladmin -uroot password root
+mysqladmin -uroot --socket=${MYSQL_UNIX_PORT} password root
 
-if [  "${mysqld_version%%%.*}" = "8.0" ]; then
-    mysql -uroot -proot <<EOF
-        ALTER USER 'db'@'%' IDENTIFIED WITH mysql_native_password BY '$MYSQL_PASSWORD';
-        ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY '$MYSQL_ROOT_PASSWORD';
-        ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '$MYSQL_ROOT_PASSWORD';
+if [[  "${mysqld_version%%%.*}" =~ ^8.[04]$ ]]; then
+    mysql -uroot -proot --socket=${MYSQL_UNIX_PORT} <<EOF
+    ALTER USER 'db'@'%' IDENTIFIED WITH caching_sha2_password BY '$MYSQL_PASSWORD';
+    ALTER USER 'root'@'%' IDENTIFIED WITH caching_sha2_password BY '$MYSQL_ROOT_PASSWORD';
+    ALTER USER 'root'@'localhost' IDENTIFIED WITH caching_sha2_password BY '$MYSQL_ROOT_PASSWORD';
```

Comment:
I think we should move this condition to other place. (it's not needed here because `caching_sha2_password` is the default.)

---

I have an old php 7.3 + mysql 8.0 project, `caching_sha2_password` is not supported there https://www.php.net/manual/en/mysqli.requirements.php

If I upgrade DDEV to this PR and run that project, the database works fine.

And I have this config set in `.ddev/mysql/my.cnf`:

```
[mysqld]
default_authentication_plugin = mysql_native_password
```

But when I try to import the database on that project using this PR, the connection doesn't work
```
Could not connect: SQLSTATE[HY000] [2054] The server requested authentication method unknown to the client
```
Because users have `caching_sha2_password`.

I know that I need to alter these users to have `mysql_native_password`, but others may not know that.

I think we should have `/start.sh` in dbserver, that looks at the current `default_authentication_plugin` and alters users using that plugin, so users don't need to figure it out themselves.

---

This is what we have in DDEV v1.24.1 for MySQL 8.0:

```
$ ddev exec "mysql -uroot -proot --execute='SELECT user, host, plugin FROM mysql.user;'"
mysql: [Warning] Using a password on the command line interface can be insecure.
+------------------+-----------+-----------------------+
| user             | host      | plugin                |
+------------------+-----------+-----------------------+
| db               | %         | mysql_native_password |
| root             | %         | mysql_native_password |
| db               | localhost | caching_sha2_password |
| mysql.infoschema | localhost | caching_sha2_password |
| mysql.session    | localhost | caching_sha2_password |
| mysql.sys        | localhost | caching_sha2_password |
| root             | localhost | mysql_native_password |
+------------------+-----------+-----------------------+
```

This is what we have in this PR for MySQL 8.0 (that's why nothing broke here before, it was already set to `caching_sha2_password`):

```
$ ddev exec "mysql -uroot -proot --execute='SELECT user, host, plugin FROM mysql.user;'"
mysql: [Warning] Using a password on the command line interface can be insecure.
+------------------+-----------+-----------------------+
| user             | host      | plugin                |
+------------------+-----------+-----------------------+
| db               | %         | caching_sha2_password |
| root             | %         | caching_sha2_password |
| db               | localhost | caching_sha2_password |
| mysql.infoschema | localhost | caching_sha2_password |
| mysql.session    | localhost | caching_sha2_password |
| mysql.sys        | localhost | caching_sha2_password |
| root             | localhost | caching_sha2_password |
+------------------+-----------+-----------------------+
```

Reply:
Something like this:

```bash
mysqld_version=$(mysqld --version | awk '{ print $3 }')

if [[ ("${mysqld_version%%%.*}" == "8.0" && "$(mysql -uroot -proot --socket=${MYSQL_UNIX_PORT} -e 'SELECT @@default_authentication_plugin;' -Ns 2>/dev/null)" == "mysql_native_password") ||
      ("${mysqld_version%%%.*}" == "8.4" && "$(mysql -uroot -proot --socket=${MYSQL_UNIX_PORT} -e 'SHOW PLUGINS;' -Ns 2>/dev/null | grep -q 'mysql_native_password'; echo $?)" == "0") ]]; then
    mysql -uroot -proot --socket=${MYSQL_UNIX_PORT} <<EOF
    ALTER USER 'db'@'%' IDENTIFIED WITH mysql_native_password BY '$MYSQL_PASSWORD';
    ALTER USER 'db'@'localhost' IDENTIFIED WITH mysql_native_password BY '$MYSQL_PASSWORD';
    ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY '$MYSQL_ROOT_PASSWORD';
    ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '$MYSQL_ROOT_PASSWORD';
EOF
fi

```

To enable `mysql_native_password` in:

MySQL 8.0:

```
[mysqld]
default_authentication_plugin = mysql_native_password
```

MySQL 8.4:

```
[mysqld]
mysql_native_password = ON
```

Reply:
Or if there is no way to have `/start.sh` or somehow configure `/docker-entrypoint.sh` for this, then we should have the instruction for using `mysql_native_password` in our docs?

Reply:
Another option:

Add a script to `dbserver`, that will alter `db` and `root` users to the specific plugin:

```bash
ddev -s db exec alter_db_user_auth_plugin.sh mysql_native_password
```

And add this hook to the docs:

```yaml
hooks:
    post-start:
        - exec: alter_db_user_auth_plugin.sh mysql_native_password
          service: db
```

Reply:
Or we can change the plugin to `mysql_native_password` for PHP 7.3 and below.